File: src/main/java/com/baulsupp/oksocial/util/WrappedRequestBody.java
package com.baulsupp.oksocial.util;

import java.io.IOException;
import okhttp3.MediaType;
import okhttp3.RequestBody;
import okio.BufferedSink;

public class WrappedRequestBody extends RequestBody {
  private final String contentType;
  private final RequestBody body;

  public WrappedRequestBody(RequestBody body, String contentType) {
    this.body = body;
    this.contentType = contentType;
  }

  @Override public MediaType contentType() {
    return MediaType.parse(contentType);
  }

  @Override public void writeTo(BufferedSink bufferedSink) throws IOException {
    body.writeTo(bufferedSink);
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/okhttp/WrappedRequestBody.kt
package com.baulsupp.oksocial.okhttp

import okhttp3.MediaType
import okhttp3.RequestBody
import okio.BufferedSink

class WrappedRequestBody(private val body: RequestBody, private val contentType: String) : RequestBody() {

  override fun contentType(): MediaType? {
    return MediaType.parse(contentType)
  }

  override fun writeTo(bufferedSink: BufferedSink) {
    body.writeTo(bufferedSink)
  }
}
